### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by muskan2532150**
- Create a class Classroom with the following:
   - @label instance variable, should be initialized in the constructor.
   - Setter and getter for @label (remember about attr_accessor).
   - Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
   - Create the has-many side (a classroom has many students).
   - Create the belongs-to side (a student belongs to a classroom).
   - Make sure that when adding a student to a classroom it also sets the classroom for the student.
  - Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Create a class Book with the following:
    - @title and @author instance variables, should be initialized in the constructor.
   - Setters and getters for instance variables (remember about attr_accessor).
- Create a class Rental with the following:
   - @date instance variable, should be initialized in the constructor.
   - Setter and getter for @date (remember about attr_accessor).
 - Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class 
        Rental.The following should be implemented:
    - Create the has-many side of Book and Rental (a book has many rentals). 
   - Create the belongs-to side of Rental and Book (a rental belongs to a book).
   - Create the has-many side of Person and Rental (a person has many rentals).
   - Create the belongs-to side of Rental and Person (a rental belongs to a person).
- Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  def can_use_services?
+    of_age? || @parent_permission
+  end
+
+  private
+
+  def of_age?
+    @gae >= 18
+  end
+end
```
Comment (BertrandConxy) [inline person.rb]: - [ ] Good job so far 👏 As a Person class has-many Rentals as a relationship/ association, you are supposed to create a method that adds rentals. As a hint, It should look like this 👇 
```ruby
  def add_rental(book, date)
    Rental.new(date, book, self)
  end

```
where `self` represents the Person object itself 😃 👍

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+    rental.book = self
+  end
```
Comment (BertrandConxy) [inline book.rb]: - [ ] Your method looks great 👏 In addition to what you have done, I can see that you pass `rental` object as an argument but also remember that in order to create a rental object, you have to provide pass in a book object. That I why I think that assigning rental a book attribute (on line 12) should come before pushing the rental object into the @rentals array (on line 11) What do you think? 
However, the best way I would suggest that you do it is to make this method accept date and person parameters and then use them to create a rental object and then push it into @rentals array all done within this function. Like this 👇 

```ruby
  def add_rental(person, date)
    rental = Rental.new(date, self, person)
   @rentals.push(rental)
  end
```

**Review by BertrandConxy — CHANGES_REQUESTED** [review]
Hi @muskan2532150, 

Good job so far! 👏 👏 
There are some issues that you still need to work on to go to the next project but you are almost there! 🏃‍♂️ 🏃‍♂️ 


## Highlights 💪 
- No linter errors
- correct flow is used
- good PR


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @BertrandConxy in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Whoistolu — APPROVED** [review]
### Hi @muskan2532150 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

### Highlight
✔️ Descriptive PR
✔️ Correct Git-flow

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request closed without merging**